crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -126,8 +129,19 @@ pub fn collect_profiling_info(
                     Ok(CoreConcreteLibfunc::FunctionCall(_))
                 ) {
                     // Push to the stack.
+                    if let Some(stack_element) = function_stack.last_mut() {
+                        if user_function_name == stack_element.caller_function_name {
+                            stack_element.recursive_calls_count += 1;
+                            continue;
+                        }
+                    }
```

Comment:
Please abstract it as stack operation `push`. Please do the same for `pop`. Recursive calls counting should be an implementation detail of the stack